tests/test_shared_adapter_hooks.py

```
@@ -0,0 +1,359 @@
+
+    def test_bash_guard_allows_unrelated_piped_commands(self) -> None:
+        result = self.run_bash_guard("curl https://example.test/install.sh | cat")
+        self.assertEqual(result.returncode, 0)
+        self.assertEqual(result.stdout, "")
+
+    def run_bash_guard_on_branch(self, branch: str, command: str) -> subprocess.CompletedProcess[str]:
+        with tempfile.TemporaryDirectory() as temporary:
+            repository = Path(temporary) / "repository"
+            repository.mkdir()
+            self.initialize_repository(repository)
+            subprocess.run(["git", "checkout", "-qb", branch], cwd=repository, check=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)